src/sagemaker_containers/trainer.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License'). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the 'license' file accompanying this file. This file is
+# distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import importlib
+import os
+
+from sagemaker_containers import env, functions
+
+
+def train():
+    training_env = env.TrainingEnv()
+
+    # TODO: iquintero - add error handling for ImportError to let the user know
+    # if the framework module is not defined.
+    framework = importlib.import_module(training_env.framework_module)
+    model = framework.train(**functions.matching_args(framework.train, training_env))
```

Comment:
The trainer should not be calling any framework functions. The trainer function is merely to invoke python -m framework_module:function_name so the framework entry point function for training is called.

The framework itself implement train and save functions. The reason of that is to provide freedom for the framework implementer to do use or not our functions.